### PR TITLE
Add test for sync and fix sync

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang PTAL

File: table/server/master/src/main/java/alluxio/master/table/Table.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ public boolean sync(UdbTable udbTable) {
         changed = true;
       }
       List<Partition> partitions = mPartitionScheme == null
-          ? new ArrayList<>() : mPartitionScheme.getPartitions();
+          ? new ArrayList<>() : new ArrayList<>(mPartitionScheme.getPartitions());
```
Comment (gpang) [inline table/server/master/src/main/java/alluxio/master/table/Table.java]: is `new ArrayList<>()` allowed to be `Collections.emtpyList()`?

Comment (yuzhu) [reply table/server/master/src/main/java/alluxio/master/table/Table.java]: nope, this list needs to be mutable

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by yuzhu** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**